modules/renter/negotiate.go

```diff
@@ -2,233 +2,328 @@ package renter
 
 import (
 	"errors"
-	"io"
+	//"io"
 	"net"
-	"os"
-	"sync/atomic"
+	"sync"
 	"time"
 
-	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/encoding"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-const (
-	defaultWindowSize = 288 // 48 Hours
-)
+// A hostUploader uploads pieces to a host. It implements the uploader interface.
+type hostUploader struct {
+	// constants
+	settings         modules.HostSettings
+	masterKey        crypto.TwofishKey
+	unlockConditions types.UnlockConditions // renter needs to save this!
```

Comment:
This looks right to me. You need to be certain which public key is which, but if you formally define that in the protocol spec (IE renter is always [0] and host is always [1]), then problem solved.